adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/CoverageRDD.scala

```diff
@@ -0,0 +1,148 @@
+    }
+  }
+
+  /**
+   * Gets coverage overlapping specified ReferenceRegion. For large, ReferenceRegions,
+   * base pairs per bin (bpPerBin) can be specified to bin together ReferenceRegions of
+   * equal size. The coverage of each bin is the mean coverage over all base pairs in that bin.
+   *
+   * @param region ReferenceRegion to fetch overlapping coverage from
+   * @param bpPerBin base pairs per bin, number of bases to combine to one bin
+   * @return RDD of Coverage Records
+   */
+  def getAggregatedCoverage(region: ReferenceRegion, bpPerBin: Int = 1): RDD[Coverage] = {
+
+    val flattenedCoverage = features2Coverage(region)
+
+    if (bpPerBin == 1) {
+      flattenedCoverage // no binning, return raw results
+    } else {
+      flattenedCoverage.map(r => {
+        // key coverage by binning start site mod bpPerbin
+        // subtract region.start to shift mod to start of ReferenceRegion
+        val key = r.start - ((r.start - region.start) % bpPerBin)
+        (key, r)
+      }).groupByKey
+        .map(r => {
+          // compute average coverage in bin
+          val averageCount = r._2.map(_.count).sum / r._2.map(_.count).size
+          Coverage(r._2.head.contigName, r._1, r._1 + bpPerBin, averageCount)
+        })
+    }
+  }
+
+  /**
+   * Filters FeatureRDD by region and flattens ReferenceRegions to individual base pair
+   * coverage.
+   * @param region Region to filter FeatureRDD
+   * @return Flattened RDD of coverage
+   */
+  private def features2Coverage(region: ReferenceRegion): RDD[Coverage] = {
+    val features = featureRdd.filterByOverlappingRegion(region)
+
+    // flatmap features in region to coverage
+    features.rdd.flatMap(r => {
+      val positions: List[Long] = List.range(r.getStart, r.getEnd)
+      positions.map(n => Coverage(r.getContigName, n, n + 1, r.getScore))
+    }).filter(r => r.start >= region.start && r.start < region.end) // filter out positions from flanking regions
+  }
+
+  /**
+   * Gets flattened RDD of coverage, with coverage at each base pair
+   * @return RDD[Coverage] from underlying FeatureRDD
+   */
+  def flattenedRdd: RDD[Coverage] = {
+    // flatmap features in region to coverage
+    featureRdd.rdd.flatMap(r => {
+      val positions: List[Long] = List.range(r.getStart, r.getEnd)
+      positions.map(n => Coverage(r.getContigName, n, n + 1, r.getScore))
+    })
+  }
+
+  /**
+   * Gets raw RDD of coverage
+   * @return RDD[Coverage] from underlying FeatureRDD
+   */
+  def rdd: RDD[Coverage] = {
+    featureRdd.rdd.map(r => {
+      Coverage(r.getContigName, r.getStart, r.getEnd, r.getScore)
+    })
+  }
+}
+
+/**
+ * Coverage Recordfor CoverageRDD
+ * @param contigName Specifies chromosomal location of coverage
+ * @param start Specifies start position of coverage
+ * @param end  Specifies end position of coverage
+ * @param count Specifies count of coverage at location
+ */
+case class Coverage(contigName: String, start: Long, end: Long, count: Double)
```

Comment:
I'm thinking `Coverage` here and `CoverageRDD` cover (sorry) your use case appropriately.

This case class and an RDD of continuous values over regions could be made more generic though, if only in name, and in turn support [wiggle](https://genome.ucsc.edu/goldenPath/help/wiggle.html)/[bigWig](https://genome.ucsc.edu/goldenPath/help/bigWig.html)/[bedGraph](https://genome.ucsc.edu/goldenPath/help/bedgraph.html) formats.  I'm not feeling very creative at the moment, there must be something better than `ContinuousValuedFeatureRDD`.  Something for a future feature request?